### PR TITLE
Domain Details Screen as WebView Enhancement

### LINKED ISSUE #19071: 🤖 Domain Details Screen as WebView
A webview that displays the domain details.

## Screenshot Reference

Image

### PR DESCRIPTION
**Pull request by antonis**
Fixes #19071

## Description
This PR enhances the implementation of https://github.com/wordpress-mobile/WordPress-Android/pull/19429 by creating the domain details url based on the type of domain (ref pc8HXX-1gR-p2).
It also moves the `DomainManagementViewModelTest` in the correct package.

## To test:
1. Enable the Domain Management feature flag (in debug settings)
2. Open the My domains screen: Me -> Domains -> My Domains
3. Tap on each domain
4. Verify that the proper domain details page opens in the webview

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
`GetDomainDetailsUrlTest`

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/domains/management/GetDomainDetailsUrlTest.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            domain = null,
+            siteSlug = "domain.slug",
+            type = "mapping"
+        )
+        val actualDetailUrl = domain.getDomainDetailsUrl()
+        assertThat(actualDetailUrl).isNull()
+    }
+
+    @Test
+    fun `WHEN the slug is null THEN the default detail url is null`() {
```
Comment (mkevins) [inline WordPress/src/test/java/org/wordpress/android/ui/domains/management/GetDomainDetailsUrlTest.kt:59]: Do we need to handle these `null` cases in the UI somehow?

Comment (antonis) [reply WordPress/src/test/java/org/wordpress/android/ui/domains/management/GetDomainDetailsUrlTest.kt:59]: Good point. I've only added a null check on the UI that makes the row not actionable. I think this is an edge case. https://github.com/wordpress-mobile/WordPress-Android/pull/19447/files#diff-e744ac2edcf6c03468d097a679972427d75f766bf80f74fb61fb97d3924693c4R5

**Review by mkevins — APPROVED** [review]
The code changes look good to me. I also tested this via the sandbox using the mocked domains flag. Unfortunately, since the `siteSlug` was `null` in those cases, I was unable to test all variations via the UI. But, there were no regressions, so I'm approving this, and we can tackle any needed UI adjustments later (if necessary, e.g. to handle the `null` cases).

**Comment by antonis** [comment]
Thank you for reviewing and testing @mkevins 🙇 

> there were no regressions, so I'm approving this, and we can tackle any needed UI adjustments later (if necessary, e.g. to handle the null cases).

Makes sense. I'll proceed with merging this PR and we can iterate with edge cases if needed 👍
